FCCee/CLD/compact/CLD_o2_v05/Beampipe_o4_v05.xml

```diff
@@ -0,0 +1,157 @@
+<lccdd>
+
+  <info name="FCCee"
+        title="FCCee low impedance Beam pipe with small radius"
+        author="Andre Sailer"
+        url="no"
+        status="development"
+        version="1.0">
+    <comment>A beampipe for FCCee detector based on CLD</comment>
+  </info>
+    
+    <!--  Definition of global dictionary constants          -->
+    <define>
+    <!--  Definition of global dictionary constants          -->
+      <constant name="beampipegoldwidth" value="BeamPipeGoldWidth"/>
+      <constant name="beampipegoldtolerance" value="BeamPipeGoldTolerance"/>
+    </define>
+
+    <!--  Definition of the used visualization attributes    -->
+    <display>
+        <vis name="BeamPipeVis" alpha="0.0" r="0.0" g="1.0" b="0.0" showDaughters="true" visible="false"/>
+        <vis name="GoldCoatingVis" alpha="0.0" r="0.0" g="1.0" b="1.0" showDaughters="true" visible="true"/>
+        <vis name="TubeVis"  alpha="1.0" r="1.0" g="0.7"  b="0.5"   showDaughters="true"  visible="true"/>
+        <vis name="VacVis"   alpha="1.0" r="1.0" g="1.0"  b="1.0"   showDaughters="true"  visible="false"/>
+    </display>
+
+    <detectors>
+
+      <!-- ;radius calculator lisp
+      (+ 12.0 (* (/ (- 28.9 12.0) (- 1190.0 90.0)) (- 110.0 90.0) ) )
+      -->
+        <comment>Part of beampipe made of AlBeMet162 and Paraffin flow</comment>
+
+        <detector name="BeBeampipe" type="DD4hep_Beampipe_o1_v01" insideTrackingVolume="true" nocore="true" vis="BeamPipeVis">
+            <parameter crossingangle="CrossingAngle" />
+            <envelope vis="BlueVis">
+                <shape type="Assembly"/>
+            </envelope>
+
+            <section type="Center"               start="0*mm"    end="CentralBeamPipe_zmax" rMin1="CentralBeamPipe_rmax"  rMin2="CentralBeamPipe_rmax"    rMax1="CentralBeamPipe_rmax+BeamPipeWidth"  rMax2="CentralBeamPipe_rmax+BeamPipeWidth"                material="CentralBeamPipeMaterial" name="VertexInnerBe" />
```

Comment:
Hi @andresailer . Could you replace this line by the following, to have one layer each for the AlBeMet162 wall, the paraffin and then the wall again? (I've tested it and it works)
```
            <!-- Inner AlBeMet162 layer -->
            <section type="Center" start="0*mm" end="CentralBeamPipe_zmax" rMin1="CentralBeamPipe_rmax" rMin2="CentralBeamPipe_rmax" rMax1="CentralBeamPipe_rmax+BeamPipeWidthWall" rMax2="CentralBeamPipe_rmax+BeamPipeWidthWall" material="AlBeMet162" name="VertexInnerAlBeMet" />

            <!-- Paraffin layer for cooling -->
            <section type="Center" start="0*mm" end="CentralBeamPipe_zmax" rMin1="CentralBeamPipe_rmax+BeamPipeWidthWall" rMin2="CentralBeamPipe_rmax+BeamPipeWidthWall" rMax1="CentralBeamPipe_rmax+BeamPipeWidthWall+BeamPipeWidthCooling" rMax2="CentralBeamPipe_rmax+BeamPipeWidthWall+BeamPipeWidthCooling" material="LiquidNDecane" name="VertexCooling" />

            <!-- Outer AlBeMet162 layer -->
            <section type="Center" start="0*mm" end="CentralBeamPipe_zmax" rMin1="CentralBeamPipe_rmax+BeamPipeWidthWall+BeamPipeWidthCooling" rMin2="CentralBeamPipe_rmax+BeamPipeWidthWall+BeamPipeWidthCooling" rMax1="CentralBeamPipe_rmax+BeamPipeWidth" rMax2="CentralBeamPipe_rmax+BeamPipeWidth" material="AlBeMet162" name="VertexOuterAlBeMet" />
```

Reply:
If a tolerance is needed between the layers, you can use the following instead:
```
            <!-- Inner AlBeMet162 layer -->
            <section type="Center" start="0*mm" end="CentralBeamPipe_zmax" rMin1="CentralBeamPipe_rmax" rMin2="CentralBeamPipe_rmax" rMax1="CentralBeamPipe_rmax+BeamPipeWidthWall" rMax2="CentralBeamPipe_rmax+BeamPipeWidthWall" material="AlBeMet162" name="VertexInnerAlBeMet" />

            <!-- Paraffin layer for cooling -->
            <section type="Center" start="0*mm" end="CentralBeamPipe_zmax" rMin1="CentralBeamPipe_rmax+BeamPipeWidthWall+BeamPipeTolerance" rMin2="CentralBeamPipe_rmax+BeamPipeWidthWall+BeamPipeTolerance" rMax1="CentralBeamPipe_rmax+BeamPipeWidthWall+BeamPipeWidthCooling" rMax2="CentralBeamPipe_rmax+BeamPipeWidthWall+BeamPipeWidthCooling" material="LiquidNDecane" name="VertexCooling" />

            <!-- Outer AlBeMet162 layer -->
            <section type="Center" start="0*mm" end="CentralBeamPipe_zmax" rMin1="CentralBeamPipe_rmax+BeamPipeWidthWall+BeamPipeWidthCooling+BeamPipeTolerance" rMin2="CentralBeamPipe_rmax+BeamPipeWidthWall+BeamPipeWidthCooling+BeamPipeTolerance" rMax1="CentralBeamPipe_rmax+BeamPipeWidth" rMax2="CentralBeamPipe_rmax+BeamPipeWidth" material="AlBeMet162" name="VertexOuterAlBeMet" />

```

Reply:
I am split (pun intended) about this. I didn't want to put different volumes to keep the tracking geometry, which puts a surface into the central beampipe, simpler. (although there already is the second one with the gold.

Reply:
Mhh, I didn't consider that. Don't you think though that it'll make some difference for physics whether these two materials (with different X0) are separated or mixed together (as it is with CentralBeampipeMaterial)?

In any case, let me know how you'd like to have this (I'll adapt what I have in my MR then accordingly)

Reply:
I put in the split beampipe, using shorter variable names though.